packages/core/src/tenants/SharedTenantContext.ts

```diff
@@ -0,0 +1,32 @@
+import type { StorageProviderData } from '@logto/schemas';
+import { storageProviderDataGuard, StorageProviderKey, Systems } from '@logto/schemas';
+import { convertToIdentifiers } from '@logto/shared';
+import type { CommonQueryMethods } from 'slonik';
+import { sql } from 'slonik';
+
+const { table, fields } = convertToIdentifiers(Systems);
+
+export default class SharedTenantContext {
```

Comment:
how about we implement a `SystemContext` like:

```ts
class SystemContext {
  static shared = new SystemContext();
}
```

thus we can leverage the [singleton pattern](https://en.wikipedia.org/wiki/Singleton_pattern) for all tenants without explicitly passing a "shared context" during app init

Reply:
in the future we can move all system actions (like checking RLS and alteration state) to this class

Reply:
updated